### PR TITLE
Ordinal Fix

### PR DESCRIPTION
**Pull request by Rajeshwari4**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -556,11 +556,14 @@ private static String toSql(RelNode root, SqlDialect dialect,
     final String bigQueryExpected = "SELECT '1' AS a, SKU + 1 AS B, '1' AS d\n"
         + "FROM foodmart.product\n"
         + "GROUP BY 1, B";
+    final String expectedSpark = "SELECT '1' a, SKU + 1 B, '1' dn"
```
Comment (deepakkadam-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:559]: Is "dn" a typo; instead should it be "d \n"
Please check

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -556,11 +556,14 @@ private static String toSql(RelNode root, SqlDialect dialect,
...
+        + "FROM foodmart.productn"
```
Comment (deepakkadam-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:560]: Is "productn" a typo; instead should it be "product \n"
Please check

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -1207,7 +1210,12 @@ private static String toSql(RelNode root, SqlDialect dialect,
         + "GROUP BY CASE WHEN CAST(salary AS DECIMAL(14, 4)) = 20 THEN MAX(salary) OVER "
         + "(PARTITION BY position_id RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) "
         + "ELSE NULL END";
-    final String expectedSpark = expectedHive;
+    final String expectedSpark = "SELECT rnkn"
+        + "FROM (SELECT CASE WHEN CAST(salary AS DECIMAL(14, 4)) = 20 THEN MAX(salary) OVER "
+        + "(PARTITION BY position_id RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) "
+        + "ELSE NULL END rnkn"
```
Comment (deepakkadam-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:1216]: Are "rnkn" and "tn" typos?

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java [hunk 4]
```diff
@@ -117,6 +121,7 @@ public enum SqlConformanceEnum implements SqlConformance {
     case LENIENT:
     case BIG_QUERY:
     case MYSQL_5:
+    case SPARK:
```
Comment (pranav-joshi-dm) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:124]: Please edit the java doc for all the parent functions (in SqlConformance) of overridden functions which are changed. Add SPARK in the function description for which function will return true.

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java [hunk 5]
```diff
@@ -131,6 +136,7 @@ public enum SqlConformanceEnum implements SqlConformance {
     case MYSQL_5:
     case PRESTO:
     case SNOWFLAKE:
+    case SPARK:
```
Comment (pranav-joshi-dm) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:139]: Can you please check if handling for SPARK is required for below functions as well : isHavingAlias(), semantics(), allowNiladicParentheses()

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (Rajeshwari4) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: It was a typo..

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (Rajeshwari4) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: Made changes

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 3 above)
Comment (Rajeshwari4) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: changed to rnk\n and t\n

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java (hunk 5 above)
Comment (Rajeshwari4) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:139]: Not in Scope..

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java (hunk 4 above)
Comment (Rajeshwari4) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:124]: Made changes according to the comment given above.

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java (hunk 5 above)
Comment (pranav-joshi-dm) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:139]: RTP-208 created for out of scope functions.

**Comment by pranav-joshi-dm** [comment]
Changes look good. L1 review completed.

File: core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java [hunk 6]
```diff
@@ -182,6 +189,7 @@ public enum SqlConformanceEnum implements SqlConformance {
     case HIVE:
     case BIG_QUERY:
     case SNOWFLAKE:
+    case SPARK:
```
Comment (pranav-joshi-dm) [inline core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:192]: As we have changes for SortBy as well, please add test case for it as well.

Comment (Rajeshwari4) [reply core/src/main/java/org/apache/calcite/sql/validate/SqlConformanceEnum.java:192]: Added TCs for SortBy..

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 7]
```diff
@@ -10627,4 +10648,15 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
         .withSpark()
         .ok(expectedSparkSql);
   }
+
+  @Test public void testSortByOrdinalForSpark() {
+    final String query = "SELECT \"product_id\" from \"product\"\n"
+        + "order by \"product_id\"";
```
Comment (pranav-joshi-dm) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:10654]: As this is test case for sort by Ordinal .. plz mention ORDER BY 1, as 1 will here point to 1st column

**Comment by pranav-joshi-dm** [comment]
Changes looks good.

**Review by deepakkadam-dm — APPROVED** [review]
Looks Ok

**Pull request merged**